File: pubmed/writePubMedProjectXML.pl
#!/usr/bin/perl

use strict;
use warnings;

use Getopt::Std;
use XML::LibXML;

use lib '../modules/perl';
require ImProjectXml;

use feature ':5.12';

my $usage = "Usage:writePubMedProjectXML.pl [-h] [-i <project-xml-path>] <data-path> <info-file-path> <taxon-id-path>

synopsis: read the taxon id file and creates an InterMine project XML <source> entry
If a <project-xml-path> is given then this entry is inserted directly into the given project XML
otherwise it is printed on stdout

options:
\t-h\tthis usage
";

my (%opts, $insert);

getopts('hi:', \%opts);
defined $opts{"h"} and die $usage;
defined $opts{"i"} and $insert++;
my $insertPath = $opts{i};

@ARGV > 2 or die $usage;

my ($dataPath, $infoFilePath, $taxonIdPath, $projectXmlPath) = @ARGV;

open TAXONS, $taxonIdPath or die "Could not open $taxonIdPath: $!";
my $taxonIds = <TAXONS>;
chomp($taxonIds);
close TAXONS;

my $sourceXml 
  = ImProjectXml::generateSource(
    "pubmed-gene", "pubmed-gene", 0, 
    [
      ['src.data.dir', 'location', $dataPath], 
      ['pubmed.organisms', $taxonIds],
      ['src.data.dir.includes', "gene2pubmed"],
      ['infoFile', $infoFilePath]
    ]);

if ($insert) {
  ImProjectXml::insertSourceIntoProjectXml($insertPath, $sourceXml);
} else {
  say $sourceXml;
}

File: pubmed/writePubMedProjectXml.py
#!/usr/bin/env python

import os
import sys

sys.path.insert(1, os.path.dirname(os.path.abspath(__file__)) + '/../modules/python')
import intermine.project as imp
import intermine.utils as imu
import synbio.dataset as sbds

############
### MAIN ###
############
parser = imu.ArgParser('Add PubMed source entry to InterMine SynBioMine project XML.')
parser.add_argument('datasetPath', help='path to the dataset location.')
parser.add_argument('pubMedDataPath', help='path to gene2pubmed file.')
parser.add_argument('-v', '--verbose', action="store_true", help="verbose output")
args = parser.parse_args()

datasetPath = args.datasetPath
ds = sbds.Dataset(datasetPath)
pubMedDataPath = args.pubMedDataPath

logPath = "%s/logs/writeOrthoDbProjectXml.log" % datasetPath
sys.stdout = imu.Logger(logPath)
projectXmlPath = "%s/intermine/project.xml" % datasetPath

imp.addSourcesToProject(
    "%s/intermine/project.xml" % datasetPath,
    [
        imp.Source(
            'pubmed-gene', 'pubmed-gene',
            [
                { 'name':'src.data.dir',            'location':os.path.dirname(pubMedDataPath) },
                { 'name':'src.data.dir.includes',   'value':'gene2pubmed' },
                { 'name':'pubmed.organisms',        'value':ds.getTaxonsAsString() }
            ]),

        imp.Source(
            'update-publications', 'update-publications',
            [
                { 'name':'src.data.file', 'location':'build/publications.xml' },
                { 'name':'loadFullRecord', 'value':'true' }
            ])
    ])